cpp/CMakeLists.txt

```diff
@@ -2,8 +2,9 @@ cmake_minimum_required(VERSION 3.19)
 set(CMAKE_POLICY_DEFAULT_CMP0077 NEW) # option respects set variables
 project(tosca)
 
-option(TOSCA_ASAN "Enable AddressSanitizer for Tosca targets.")
-option(TOSCA_ASSERT "Enable Tosca specific assertions.")
+option(TOSCA_ASAN "Enable AddressSanitizer for Tosca targets." OFF)
+option(TOSCA_ASSERT "Enable Tosca specific assertions." ON)
```

Comment:
The assertions should be off in the release build (target `tosca-cpp`). Is it like that?

Reply:
Same defaults as in makefile, makefile values take precedence. 
![image](https://github.com/Fantom-foundation/Tosca/assets/170023705/557421b4-c005-4c4a-9df4-43663cc0ab6c)
if this is wrong, let me know.

Reply:
We should discuss the role of assertions in the C++ code offline some time and then update those flags if needed. So, its beyond the scope of this PR.